lib/segment/src/index/field_index/mmap_point_to_values.rs

```diff
@@ -0,0 +1,389 @@
+}
+
+impl<'a> MmapValue<'a> for IntPayloadType {
+    fn mmaped_size(&self) -> usize {
+        std::mem::size_of::<Self>()
+    }
+
+    fn read_from_mmap(bytes: &[u8], offset: usize) -> OperationResult<Self> {
+        Self::read_from_prefix(&bytes[offset..])
+            .ok_or_else(|| OperationError::service_error(NOT_ENOUGHT_BYTES_ERROR_MESSAGE))
+    }
+
+    fn write_to_mmap(&self, bytes: &mut [u8], offset: usize) -> OperationResult<()> {
+        self.write_to_prefix(&mut bytes[offset..])
+            .ok_or_else(|| OperationError::service_error(NOT_ENOUGHT_BYTES_ERROR_MESSAGE))
+    }
+}
+
+impl<'a> MmapValue<'a> for FloatPayloadType {
+    fn mmaped_size(&self) -> usize {
+        std::mem::size_of::<Self>()
+    }
+
+    fn read_from_mmap(bytes: &[u8], offset: usize) -> OperationResult<Self> {
+        Self::read_from_prefix(&bytes[offset..])
+            .ok_or_else(|| OperationError::service_error(NOT_ENOUGHT_BYTES_ERROR_MESSAGE))
+    }
+
+    fn write_to_mmap(&self, bytes: &mut [u8], offset: usize) -> OperationResult<()> {
+        self.write_to_prefix(&mut bytes[offset..])
+            .ok_or_else(|| OperationError::service_error(NOT_ENOUGHT_BYTES_ERROR_MESSAGE))
+    }
+}
+
+impl<'a> MmapValue<'a> for GeoPoint {
+    fn mmaped_size(&self) -> usize {
+        2 * std::mem::size_of::<f64>()
+    }
+
+    fn read_from_mmap(bytes: &[u8], offset: usize) -> OperationResult<Self> {
+        Ok(Self {
+            lon: f64::read_from_prefix(&bytes[offset..])
+                .ok_or_else(|| OperationError::service_error(NOT_ENOUGHT_BYTES_ERROR_MESSAGE))?,
+            lat: f64::read_from_prefix(&bytes[offset + std::mem::size_of::<f64>()..])
+                .ok_or_else(|| OperationError::service_error(NOT_ENOUGHT_BYTES_ERROR_MESSAGE))?,
+        })
+    }
+
+    fn write_to_mmap(&self, bytes: &mut [u8], offset: usize) -> OperationResult<()> {
+        self.lon
+            .write_to_prefix(&mut bytes[offset..])
+            .ok_or_else(|| OperationError::service_error(NOT_ENOUGHT_BYTES_ERROR_MESSAGE))?;
+        self.lat
+            .write_to_prefix(&mut bytes[offset + std::mem::size_of::<f64>()..])
+            .ok_or_else(|| OperationError::service_error(NOT_ENOUGHT_BYTES_ERROR_MESSAGE))
+    }
+}
+
+impl<'a> MmapValue<'a> for &'a str {
+    fn mmaped_size(&self) -> usize {
+        self.as_bytes().len() + std::mem::size_of::<u32>()
+    }
+
+    fn read_from_mmap(bytes: &'a [u8], offset: usize) -> OperationResult<&'a str> {
+        let bytes = &bytes[offset..];
+        let size = u32::read_from_prefix(bytes)
+            .ok_or_else(|| OperationError::service_error(NOT_ENOUGHT_BYTES_ERROR_MESSAGE))?
+            as usize;
+
+        let bytes = &bytes[std::mem::size_of::<u32>()..];
+        if bytes.len() < size {
+            return Err(OperationError::service_error(
+                NOT_ENOUGHT_BYTES_ERROR_MESSAGE,
+            ));
+        }
+
+        let bytes = &bytes[..size];
```

Comment:
(see the comment above in this file)
```suggestion
        let bytes = bytes.get(
            size_of::<u32>()..size_of::<u32>()+size,
        ).ok_or(OperationError::…)?;
```